### PR TITLE
Staggered Indexing

### PR DESCRIPTION
**Pull request by willronchetti**
- Uses new snovault feature allowing us to reindex type-by-type, minimizing downtime substantially
- Bumps GA build machine
- SPC 0.1.57
- Allow utils ^6.0.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "@hms-dbmi-bgm/react-workflow-viz": "0.1.5",
     "@gmod/tabix": "^1.5.3",
     "@gmod/vcf": "^5.0.6",
-    "@hms-dbmi-bgm/shared-portal-components": "git+https://github.com/4dn-dcic/shared-portal-components#0.1.52",
+    "@hms-dbmi-bgm/shared-portal-components": "git+https://github.com/4dn-dcic/shared-portal-components#0.1.57",
```
Comment (netsettler) [inline package.json:55]: Some reason this can't use the simpler version notation?

File: deploy/docker/production/entrypoint_deployment.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ fi
 
 # Create mapping
 # Force wipe of ES
-poetry run create-mapping-on-deploy production.ini --app-name app --wipe-es
+poetry run create-mapping-on-deploy production.ini --app-name app --staggered
```
Comment (netsettler) [inline deploy/docker/production/entrypoint_deployment.sh:18]: So this elimnates the nees for wipe-es, or it implies it, or it's just incompatible with it?

File: src/encoded/commands/create_mapping_on_deploy.py [hunk 3]
```diff
@@ -94,15 +95,19 @@ def main():
...
 
     args = parser.parse_args()
     app = get_app(args.config_uri, args.app_name)
     # Loading app will have configured from config file. Reconfigure here:
     set_logging(in_prod=app.registry.settings.get('production'), log_name=__name__, level=logging.DEBUG)
     # set_logging(app.registry.settings.get('elasticsearch.server'), app.registry.settings.get('production'),
     #             level=logging.DEBUG)
-
-    _run_create_mapping(app, args)
+    if not args.staggered:
```
Comment (netsettler) [inline src/encoded/commands/create_mapping_on_deploy.py:107]: swap the order of the clauses and remove the `not`?

File: src/encoded/commands/create_mapping_on_deploy.py [hunk 4]
```diff
@@ -94,15 +95,19 @@ def main():
...
+        _run_create_mapping(app, args)
+    else:
+        reindex_by_type_staggered(app)
```
Comment (netsettler) [inline src/encoded/commands/create_mapping_on_deploy.py:110]: Given this doesn't run run_create_mapping (or does it do it inside) should it be checking that the args are not supplied? If someone is giving specialized args, the data flow will be dropped.

**Review by netsettler — APPROVED** [review]
My comments are not per se blocking, but might suggest problems that do need to be fixed.

File: src/encoded/commands/create_mapping_on_deploy.py (hunk 4 above)
Comment (willronchetti) [reply src/encoded/commands/create_mapping_on_deploy.py:110]: Added a comment on this

**Pull request merged**